addons/binding/org.openhab.binding.feed/src/main/java/org/openhab/binding/feed/handler/FeedHandler.java

```diff
@@ -0,0 +1,290 @@
+/**
+ * Copyright (c) 2014 openHAB UG (haftungsbeschraenkt) and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.feed.handler;
+
+import static org.openhab.binding.feed.FeedBindingConstants.*;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.math.BigDecimal;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.net.URLConnection;
+import java.text.DateFormat;
+import java.util.Calendar;
+import java.util.Date;
+import java.util.GregorianCalendar;
+import java.util.List;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+import org.eclipse.smarthome.config.core.Configuration;
+import org.eclipse.smarthome.core.library.types.DateTimeType;
+import org.eclipse.smarthome.core.library.types.DecimalType;
+import org.eclipse.smarthome.core.library.types.StringType;
+import org.eclipse.smarthome.core.thing.Channel;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.RefreshType;
+import org.eclipse.smarthome.core.types.State;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.rometools.rome.feed.synd.SyndEntry;
+import com.rometools.rome.feed.synd.SyndFeed;
+import com.rometools.rome.io.FeedException;
+import com.rometools.rome.io.SyndFeedInput;
+
+/**
+ * The {@link FeedHandler } is responsible for handling commands, which are
+ * sent to one of the channels and for the regular updates of the feed data.
+ *
+ * @author Svilen Valkanov - Initial contribution
+ */
+public class FeedHandler extends BaseThingHandler {
+
+    private Logger logger = LoggerFactory.getLogger(FeedHandler.class);
+
+    private String urlString;
+    private BigDecimal refreshTime;
+    private ScheduledFuture<?> refreshTask;
+    private SyndFeed currentFeedState;
+
+    public FeedHandler(Thing thing) {
+        super(thing);
+        currentFeedState = null;
+    }
+
+    @Override
+    public void initialize() {
+        checkConfiguration();
+        super.initialize();
```

Comment:
I'd think you should call startAutomaticRefresh() already here - to make sure that the Thing is working and REALLY online, because maybe you cannot at all access it.